include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -224,10 +224,11 @@ class __kernel_compiler
     __compile(_Exec&& __exec)
     {
         sycl::program __program(__exec.queue().get_context());
-        if constexpr (sizeof...(_KernelNames) > 1)
-            return __kernel_array_type{__program.build_with_kernel_type<_KernelNames>(), __program.template get_kernel<_KernelNames>())...};
-        else
-            return __program.build_with_kernel_type<_KernelNames>(), __program.template get_kernel<_KernelNames>();
+
+        using __return_type =
+            typename std::conditional<(sizeof...(_KernelNames) > 1), __kernel_array_type, sycl::kernel>::type;
+        return __return_type{
+            (__program.build_with_kernel_type<_KernelNames>(), __program.template get_kernel<_KernelNames>())...};
```

Comment:
```suggestion
            (__program.build_with_kernel_type<_KernelNames>(), __program.get_kernel<_KernelNames>())...};
```
```template``` keyword seems redundant. Please see [example](http://cmplrexplorer.intel.com/z/WWe4hM). Line 22 requires `template` keyword while line 25 doesn't